Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -264,7 +283,20 @@ - (void)startLoading {
         NSError* error = nil;
 
         try {
-            auto httpRequestMessage = _requestMessageForNS(self.request);
+            auto httpRequestMessage = _requestMessageForNS(self.request, _flattenedBodyStream);
+
+            // VSO 6693048: If this properly supports streaming bodies instead of flattening the data up front,
+            // more accurate callbacks can be done
+            NSInteger contentSize = self.request.HTTPBody ? self.request.HTTPBody.length : _flattenedBodyStream.size();
+            if (contentSize > 0) {
+                __dispatchClientCallback(self, ^void() {
+                    [self.client URLProtocol:self
```

Comment:
This won't actually send until the message is sent, since it's dispatching to the client thread. The other option would be to put this in _asyncHttpOperation:completedWithStatus:, but that seems too _late_...